jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/admin/commands/AddPeopleAssignmentsCommand.java

```diff
@@ -56,7 +62,7 @@ public Void execute(Context cntxt) {
         
         Task task = context.getTaskQueryService().getTaskInstanceById(taskId);       
         // security check
-        if (!isBusinessAdmin(userId, task.getPeopleAssignments().getBusinessAdministrators(), context)) {
+        if (!skipOwnerCheck && !isBusinessAdmin(userId, task.getPeopleAssignments().getBusinessAdministrators(), context)) {
```

Comment:
We were discussing this with @sutaakar and realized that the same property is now used for 2 slightly different things. It can be set for example for getting tasks assigned as a potential user, in which case a userId has to be provided and this will be used for the query. Or the same can be used for example for comments, when even when using the bypass we have to provide the correct user for that particular case, otherwise the security check for the case will fail. So it can be said that in both these examples we can bypass the authentication by providing a different, but always valid user with regards to the context of the operation. So bypass in these cases just allows users to behave as another user, but still a valid one.

However, here it seems it doesn't really matter which user we provide, the only important thing is if the `skipOwnerCheck` is true, which means if bypass is activated. In that case we can completely bypass the authentication without knowing any context of the operation. So we can even provide "non-existing-user" and the operation would succeed. Wouldn't it be better to use the same approach as we have used until now? That means that even with bypass activated a user would have to provide a valid user for this context, so they would have to provide the correct business administrator user. So bypass actually allows to "impersonate" another, still valid user, in that case a business administrator. What do you think?

Also I noticed that actually the passed `userId` doesn't seem to be used at all when a bypass is turned on in the current implementation.

Reply:
the current user ends up in the TaskEventImpl. See test case.
Removed the skipping flag.